### PR TITLE
Validation Support

### LINKED ISSUE #79: Capability to validate payload
As the title says.

We are going a build an ambitious BE/FE collaboration flow / toolchain which depends on this feature. Stay tuned!

### PR DESCRIPTION
**Pull request by Cryrivers**
Close #79

Add a new CLI option `--outputFile` which outputs the compiled file (and corresponding type declaration) only.

## Usage
```ts
// interface.ts
export type GetItemRequest = {
  productId: number;
}
``` 

### Compile `interface.ts`
```bash
ms --configFile interface.ts --outputFile interface.js
```

### Install dependencies and copy over `interface.js` and `interface.d.ts`
```bash
yarn add @manta-style/runtime @manta-style/core
```

### Use it
```js
import { GetItemRequest } from './interface';
import { EmptyContext } from '@manta-style/core';

async function validateUserInput(value) {
  return await GetItemRequest.validate(value, EmptyContext);
}
```

**Now you get runtime validation that respects to your type definition**

TODO (not in this PR):
- [ ] Remove the param `context` in `validate` (with some trade-offs)
- [ ] Make `validate` synchronous

Note: This PR is blocking our full-stack scaffolding tool based on Egg.js with E2E type coverage support.

### CODE REVIEW

**Comment by Cryrivers** [comment]
Sorry this PR is still WIP. Will add reviewer later.

File: packages/core/runtime/src/types/NullableType.ts [hunk 1]
```diff
@@ -21,6 +22,13 @@ export default class NullableType extends Type {
   ) {
     return this.nullableType.deriveLiteral(annotations, context);
   }
+  public async validate(value: unknown, context: MantaStyleContext) {
+    return (
+      this.type.validate(value, context) ||
+      MantaStyle.UndefinedKeyword.validate(value) ||
+      MantaStyle.NullKeyword.validate(value)
```
Comment (tanhauhau) [inline packages/core/runtime/src/types/NullableType.ts]: how come .validate can be used as `static`?

File: packages/builders/builder-typescript/src/transformer/index.ts [hunk 2]
```diff
@@ -6,13 +6,32 @@ import {
   MANTASTYLE_HELPER_NAME,
   HELPER_PACKAGE_NAME,
 } from './constants';
+import * as fs from 'fs';
+import * as path from 'path';
```
[the remark this answers is not in the document]
Comment (Cryrivers) [reply packages/builders/builder-typescript/src/transformer/index.ts]: Should be the same if `moduleTarget` in `tsconfig.json` is `commonjs`.

File: packages/core/runtime/src/types/NullableType.ts (hunk 1 above)
Comment (Cryrivers) [reply packages/core/runtime/src/types/NullableType.ts]: `UndefinedKeyword`, `NullKeyword` and other keywords are instances.

File: packages/core/runtime/src/utils/assignable.ts [hunk 3]
```diff
@@ -108,10 +108,12 @@ export async function isAssignable(
...
 
-async function everyPromise(array: Promise<boolean>[]): Promise<boolean> {
+export async function everyPromise(
+  array: Promise<boolean>[],
+): Promise<boolean> {
   return (await Promise.all(array)).every((t) => t);
 }
 
-async function somePromise(array: Promise<boolean>[]): Promise<boolean> {
+export async function somePromise(array: Promise<boolean>[]): Promise<boolean> {
```
[the remark this answers is not in the document]
Comment (Cryrivers) [reply packages/core/runtime/src/utils/assignable.ts]: Is it safe to resolve a promise multiple times?

**Pull request merged**